bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitCharacteristicFactory.java

```diff
@@ -214,6 +214,28 @@ private static int getIntFromItem(HomekitTaggedItem taggedItem) {
         return value;
     }
 
+    /** special method for tilts. it converts percentage to angle */
+    private static int getAngleFromItem(HomekitTaggedItem taggedItem) {
+        int value = 0;
+        final State state = taggedItem.getItem().getState();
+        if (state instanceof PercentType) {
+            value = (int) ((((PercentType) state).intValue() * 90.0) / 50.0 - 90.0);
+        } else if (state instanceof DecimalType) {
+            value = ((DecimalType) state).intValue();
+        } else if (state instanceof UnDefType) {
+            logger.debug("Item state {} is UNDEF {}.", state, taggedItem.getName());
+        } else {
+            logger.warn(
+                    "Item state {} is not supported for {}. Only PercentType and DecimalType (0/100) are supported.",
+                    state, taggedItem.getName());
+        }
+        return value;
```

Comment:
Maybe simplify to reuse the original method?
```suggestion
        final State state = taggedItem.getItem().getState();
        if (state instanceof PercentType) {
            return (int) ((((PercentType) state).intValue() * 90.0) / 50.0 - 90.0);
        } else {
            return getIntSupplier(taggedItem);
        }
```

Reply:
good idea. done.